### PR TITLE
Enable Vector Loads

### PR DESCRIPTION
**Pull request by micmelesse**
This PR fixes an issue where blocks where not being loaded when they exceded a certain block size. It ends up enabling Vector loads.

### CODE REVIEW

File: lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp [hunk 1]
```diff
@@ -1217,24 +1217,20 @@ struct StoreOpConversion
...
-      if (elem.getType().getIntOrFloatBitWidth() <= 8)
-        elem = rewriter.create<LLVM::SExtOp>(loc, type::i8Ty(ctx), elem);
-      Type u32Ty = typeConverter->convertType(type::u32Ty(ctx));
-      llWord =
-          insert_element(wordTy, llWord, elem,
-                         rewriter.create<LLVM::ConstantOp>(
-                             loc, u32Ty, IntegerAttr::get(u32Ty, 0)));
-      llWord = bitcast(llWord, valueElemTy);
-      store(llWord, ptrElems[vecStart]);
+      for (size_t wordIdx = 0; wordIdx < nWords; ++wordIdx) {
```
Comment (rsanthanam-amd) [inline lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp:1231]: maybe move the store call up into the other loop so something like:

      SmallVector<std::pair<Value, std::string>> asmArgs;
      for (size_t wordIdx = 0; wordIdx < nWords; ++wordIdx) {
        // llWord is a width-len composition
        Value llWord = rewriter.create<LLVM::UndefOp>(loc, wordTy);
        // Insert each value element to the composition
        for (size_t elemIdx = 0; elemIdx < wordNElems; ++elemIdx) {
          const size_t elemOffset = vecStart + wordIdx * wordNElems + elemIdx;
          assert(elemOffset < valueElems.size());
          Value elem = valueElems[elemOffset];
          if (elem.getType().isInteger(1))
            elem = rewriter.create<LLVM::SExtOp>(loc, type::i8Ty(ctx), elem);
          elem = bitcast(elem, valueElemTy);

          Type u32Ty = typeConverter->convertType(type::u32Ty(ctx));
          llWord =
              insert_element(wordTy, llWord, elem,
                             rewriter.create<LLVM::ConstantOp>(
                                 loc, u32Ty, IntegerAttr::get(u32Ty, elemIdx)));
        }
#ifdef USE_ROCM
        llWord = bitcast(llWord, valueElemTy);
        store(llWord, ptrElems[vecStart + wordIdx]);
#else

        llWord = bitcast(llWord, valArgTy);
        std::string constraint =
            (width == 64) ? "l" : ((width == 32) ? "r" : "c");
        asmArgs.emplace_back(llWord, constraint);
#endif
      }
#ifndef USE_ROCM

File: lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp [hunk 2]
```diff
@@ -1006,10 +1006,10 @@ struct LoadOpConversion
       assert(wordNElems * nWords * numVecs == numElems);
 
 #ifdef USE_ROCM
-      // TODO: need to revisit this when implementing vectorized loads
-      Value ret = load(ptrElems[vecStart]);
-      ret = bitcast(ret, valueElemTy);
-      loadedVals.push_back(ret);
+      for (size_t wordIdx = 0; wordIdx < nWords; ++wordIdx) {
+        Value ret = load(ptrElems[vecStart + wordIdx]);
+        loadedVals.push_back(ret);
```
Comment (rsanthanam-amd) [inline lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp:1011]: i think we need a bitcase here:

ret = bitcast(ret, valueElemTy);

File: lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp (hunk 1 above)
Comment (micmelesse) [reply lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp]: Done

File: lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp (hunk 2 above)
Comment (micmelesse) [reply lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp:1011]: Done

**Review by rsanthanam-amd — APPROVED** [review]

**Pull request merged**